pkg/clients/ovirt/impl.go

```diff
@@ -67,7 +71,16 @@ func (is *ovirtClient) CreateVMByMachine(
 						Threads(int64(providerSpec.CPU.Threads))))
 		}
 		if providerSpec.MemoryMB > 0 {
-			vmBuilder.Memory(int64(math.Pow(2, 20)) * int64(providerSpec.MemoryMB))
+			vmBuilder.Memory(int64(BytesInMB) * int64(providerSpec.MemoryMB))
+		}
+		if providerSpec.GuaranteedMemoryMB > 0 {
+			memoryPolicyBuilder := ovirtsdk.NewMemoryPolicyBuilder()
+			memoryPolicyBuilder.Guaranteed(int64(BytesInMB) * int64(providerSpec.GuaranteedMemoryMB))
+			memoryPolicy, err := memoryPolicyBuilder.Build()
+			if err != nil {
+				return nil, err
```

Comment:
Please wrap the error and provider context to the failure

Reply:
done